presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -252,47 +253,47 @@ protected RoleGrant computeNext()
         });
     }
 
-    public static boolean isRoleApplicable(SemiTransactionalHiveMetastore metastore, PrestoPrincipal principal, String role)
+    public static boolean isRoleApplicable(SemiTransactionalHiveMetastore metastore, ConnectorIdentity identity, PrestoPrincipal principal, String role)
```

Comment:
nit: I would recommend passing the `MetastoreContext` directly to the methods in this class

Reply:
in the existing functions, like listEnabledPrincipals, listEnabledTablePrivileges and other functions in this class, it already refers ConnectorIdentity in other purposes.